mobly/controllers/android_device_lib/jsonrpc_client_base.py

```diff
@@ -214,7 +217,7 @@ def connect(self, port, addr='localhost', uid=UNKNOWN_UID,
                     logging.exception("Failed to create socket connection!")
                     raise
                 time.sleep(1)
-
+        self.port = port
```

Comment:
The port is on whatever addr is. The client doesn't know or care if the port is local or not as long as it can connect to addr:port.